atsha204a/src/main.c

```diff
@@ -127,6 +127,30 @@ int main(int argc, char *argv[])
 	CHECK_RES("hmac", ret, buf, HMAC_LEN);
 #endif
 
+	printf("\n - SHA \n");
+	{
+		/* The caller is required to pass the padding and length
+		 * bytes of the message (Sect. 13.1)
+		 */
+		uint8_t sha_in[64] = {
+			/* Message */
+			0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+			0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF,
+			0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+			0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF,
+			/* Padding */
+			0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			/* Length */
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00
+		};
+		ret = cmd_sha(ioif, 0, NULL, buf, 1);
```

Comment:
Is this right? Function is
```c
int cmd_sha(struct io_interface *ioif, uint8_t *in, size_t in_size, uint8_t *out, size_t out_size)
```
You are giving `0` for the pointer and `NULL` for the size. Should it be the opposite? `NULL` and `0`.